js-core/homeworks/homework-6/src/main.js

```diff
@@ -0,0 +1,502 @@
+    if(boolean === true){
+      return reversToStr.reverse().join(' ');
+    }
+    return reversToStr.join(' ');
+  };
+
+console.log(reverseEachWord('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM', true));
+//MOD eht fo snoitatnemelpmi tnereffid evah sresworb tnereffiD .MOD eht gnisu nigeb ot laiceps gnihtyna od ot evah t'nod uoY
+console.log(reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents', true));
+// stnemucod LMX dna LMTH rof ecafretni gnimmargorp a si )MOD( ledoM tcejbO tnemucoD ehT
+
+console.log(reverseEachWord('Hi my Name is', false));
+// // iH ym emaN si
+
+// /* TASK 3
+//  * Посчитайте сколько одинаковых слов в предложении.
+//  * Функция должна возвращать объект у которого свойства
+//  * это повторяющиеся слово,
+//  * а значение это количество повторений
+//  * */
+
+// // Both - Java - and - Java - Script - is - programming - and - programming - OOPBased
+
+function wordCounter(sentence) {
+  let strToArr = sentence.split(' ');
+  let obj = strToArr.reduce(function(newValue, elem, index, arr) {
+    let firstElem = elem;
+    let counter = 0;
+    strToArr.forEach(function(elem, index, arr) {
+      if (firstElem === elem) {
+        counter++;
+        newValue[firstElem] = counter;
+      };
+    });
+    return newValue;
+  }, {});
+  return obj;
+};
+
+console.log(
+  wordCounter('Both Java and Java Script is programming and programming OOPBased Language'),
+);
+// /*
+//  {
+//  Both:1,
+//  Java:2,
+//  and:2,
+//  Script: 1,
+//  is: 1
+//  programming: 2
+//  OOPBased:1,
+//  Language:1
+//  }
+//  */
+
+console.log(wordCounter('asd qwe asd'));
+// /*
+//  {
+//  asd:2
+//  qwe:1
+//  }
+//  * */
+
+console.log(wordCounter('url http url www url http'));
+
+// /*
+//  {
+//  http:2,
+//  url:3,
+//  www:1
+//  }
+//  * */
+
+// /*
+//   TASK 4
+//  // Функция принимает массив у которого есть свойста _id и Company.
+//  // верните объект, у которого ключ это _id, а значение Company
+//  */
+
+function createHashTags(arr) {
+  let obj = arr.reduce(function(newValue, elem){
+    let companyName = elem.company;
+    let companyId = elem._id;
+      newValue[companyId] = companyName
+    return newValue
+  }, {});
+  return obj
+}
+
+
+console.log(createHashTags(listOfCompanys));
+// //{"584babb6eeb4137cf14c37a3":"ASIMILINE", '584babb6eeb4137cf14c37a3':'Company2', }
+
+// // @ SUPER
+// /*
+//  *
+//  * TASK 1
+//  * Выведите уникальные значения
+//  *
+//  * */
+
+// function uniqueElements(arr) {
+
+// }
+
+// //
+// let notUniqArray = [1, 1, 2, 2, 2, 5, 10, 25, 30, 5, 1, 0, 22, 3, 10, 3];
+// //
+// console.log(uniqueElements(notUniqArray)); //1,2,5,10,25,30,0,22,3,
+// console.log(uniqueElements([1, 1, 2, 3, 3])); // 1,2,3
+
+// /*
+// *
+// * super2
+// *
+// * implement array method filter function
+// *
+// *
+// * */
+
+function filter (arr, callback, thisArg) {
+  let results = [];
```

Comment:
what the third argument mean? 
could you please explain why we need it?

Can you rewrite existing solution without third argument ?